linux/src/uuid_linux.c

```diff
@@ -35,3 +36,45 @@ int uuid_produce(UUID_T destination)
     }
     return result;
 }
+
+bool is_uuid_nil(const UUID_T uuid_value)
+{
+    bool result;
+    // Codes_SRS_UUID_LINUX_11_001: [ if uuid_value is NULL then is_uuid_nil shall fail and return false. ]
+    if (uuid_value == NULL)
+    {
+        LogError("invalid argument UUID_T uuid_value=%p", uuid_value);
+        result = false;

```

Comment:
Doesn't this mean that a `NULL` `uuid_value` is not `nil`? From the point of view of the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment), should we return `true` here?

Reply:
I thought about this a lot, even thought about returning an enum.  I get what you're saying.  Do you think I should just return an enum?